src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java

```diff
@@ -146,38 +153,35 @@ public void getAllPets_whenNoPets_returnsEmptyList() {
     public void updatePet_whenPetExists_returnsUpdatedPet() {
         // given
         final var fakePetRequest = fakeDataGenerator.generateFakePetRequest();
-        final var fakePetId = fakeDataGenerator.generateRandomInteger();
         final var fakeOriginalPet = fakeDataGenerator.generateFakePet();
-        final var fakeUpdatedPet = fakeOriginalPet.toBuilder()
-                .name(fakePetRequest.getName())
-                .type(fakePetRequest.getType())
-                .age(fakePetRequest.getAge())
-                .gender(fakePetRequest.getGender())
-                .build();
+        final var fakeUpdatedPet = fakeDataGenerator.generateFakePet(fakeOriginalPet);
         final var fakePetResponse = fakeDataGenerator.generateFakePetResponse();
-        given(mockMapper.map(fakePetRequest, Pet.class)).willReturn(fakeUpdatedPet);
-        given(mockPetRepository.findById(fakePetId))
+        given(mockMapper.map(fakePetRequest, Pet.class))
+                .willReturn(fakeUpdatedPet);
+        given(mockPetRepository.findById(fakeOriginalPet.getId()))
                 .willReturn(Optional.of(fakeOriginalPet));
         given(mockPetRepository.save(fakeOriginalPet))
                 .willReturn(fakeUpdatedPet);
-        given(mockMapper.map(fakeUpdatedPet, PetResponse.class)).willReturn(fakePetResponse);
+        given(mockMapper.map(fakeUpdatedPet, PetResponse.class))
+                .willReturn(fakePetResponse);
 
         // when
-        final var actual = serviceUnderTest.updatePet(fakePetRequest, fakePetId);
+        final var actual = serviceUnderTest.updatePet(fakePetRequest, fakeOriginalPet.getId());
 
         // then
         verify(mockPetRepository).save(petCaptor.capture());
         assertThat(petCaptor.getValue()).usingRecursiveComparison().isEqualTo(fakeUpdatedPet);
         assertThat(actual)
                 .usingRecursiveComparison()
                 .isEqualTo(fakePetResponse);
+
     }
 
     @Test
     public void updatePet_whenPetNotExists_throwEntityNotFoundException() {
         // given
         final var fakePetRequest = fakeDataGenerator.generateFakePetRequest();
-        final var fakePetId = fakeDataGenerator.generateRandomInteger();
+        final var fakePetId = fakeDataGenerator.generateFakePet().getId();
```

Comment:
is it necessary to have the fake pet object?
Was this change necessary?

Reply:
It's sort of an apples to apples change. Basically doing the same thing so not _necessarily_ critical to change.